backendImpl/src/main/java/com/virtuslab/gitmachete/backend/impl/GitMacheteRepository.java

```diff
@@ -774,4 +781,41 @@ IGitMacheteRepositorySnapshot discoverGitMacheteRepository() throws GitMacheteEx
     }
 
   }
+
+  @Getter
+  private static class GitMacheteNonRootBranchCreationResult {
+    private final List<GitMacheteNonRootBranch> createdBranches;
+    private final List<String> notCreatedBranchNames;
+
+    private GitMacheteNonRootBranchCreationResult() {
+      createdBranches = List.empty();
+      notCreatedBranchNames = List.empty();
+    }
+
+    public static GitMacheteNonRootBranchCreationResult empty() {
+      return new GitMacheteNonRootBranchCreationResult();
+    }
+
+    GitMacheteNonRootBranchCreationResult(GitMacheteNonRootBranchCreationResult prevResult, String notCreatedBranch) {
```

Comment:
TBH I'd rather replace this c'tor with an instance method `withExtraSkippedBranch(String skippedBranch)`; this will look more natural at the call site (line 364)